conftest.py

```diff
@@ -8,27 +8,55 @@
 
 
 @pytest.fixture(scope="session", autouse=True)
-def redis_container():
-    # Set the default Redis version if not already set
+def redis_container(request):
+    """
+    Create a unique Compose project for each xdist worker by setting
+    COMPOSE_PROJECT_NAME. That prevents collisions on container/volume names.
+    """
+    # In xdist, the config has "workerid" in workerinput.  For the main (non-xdist)
+    # process, 'workerid' is often 'master' or something similar.
+    worker_id = request.config.workerinput.get("workerid", "master")
+
+    # Set the Compose project name so containers do not clash across workers
+    os.environ["COMPOSE_PROJECT_NAME"] = f"redis_test_{worker_id}"
     os.environ.setdefault("REDIS_VERSION", "edge")
 
-    compose = DockerCompose("tests", compose_file_name="docker-compose.yml", pull=True)
+    compose = DockerCompose(
+        context="tests",
+        compose_file_name="docker-compose.yml",
+        pull=True,
+    )
+
     compose.start()
 
+    # If you mapped the container port 6379:6379 in docker-compose.yml,
+    # you might have collisions across workers. If you rely on ephemeral
+    # host ports, remove the `ports:` block in docker-compose.yml and do:
     redis_host, redis_port = compose.get_service_host_and_port("redis", 6379)
-    redis_url = f"redis://{redis_host}:{redis_port}"
-    os.environ["REDIS_URL"] = redis_url
```

Comment:
Should line 35 also be commented out? (I added a suggestion for it.)

This isn't a blocker for me, but ideally, the codebase should only support and suggest behavior that's configurable at runtime through environment variables or CLI flags, without requiring developers to comment or uncomment code.

Reply:
I think this is just hangover code I still need to clean up. We don't need to extract the host/port here because DockerCompose via testcontainers is handling the ephemeral port mapping and the `redis_url` fixture below attaches to it when called upon to get the right host/port combination